### PR TITLE
ref: aggregate workspace lints in root Cargo.toml

### LINKED ISSUE #25: Aggregate workspace lints in root `Cargo.toml`
We currently have to define global lint configuration in each crate separately, this is a bit of a pain to maintain.

```rust
#![warn(missing_docs, unreachable_pub, rust_2021_compatibility)]
#![warn(clippy::all, clippy::pedantic)]
```

We should keep this configuration in one place, namely the root `Cargo.toml` by leveraging the `[workspace.lints]` table, like [this](https://github.com/paradigmxyz/reth/blob/e087e7ef8099042634be40cd171d6e977ca6035d/Cargo.toml#L85).

### PR DESCRIPTION
**Pull request by developeruche**
Made an update on this root Cargo.toml file introduction `rust_lint` and `clippy_lint`

Resolves #25 

#### PR Checklist

- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Cargo.toml
Comment (alexfertel) [inline Cargo.toml:55]: I think `pub_underscore_fields` and `module_name_repetitions` we will want to allow on a case-by-case basis, so it's fine not to include here.

```suggestion
[workspace.lints.rust]
rust_2021_compatibility = "warn"
missing_docs = "warn"
unreachable_pub = "warn"

[workspace.lints.clippy]
pedantic = "warn"
all = "warn"
```

File: Cargo.toml
Comment (alexfertel) [inline Cargo.toml:44]: Let's keep this blank line since it groups away alloy dependencies.

File: docs/yarn.lock
Comment (alexfertel) [inline docs/yarn.lock:1]: Let's not commit this file to source control yet.

**Review by alexfertel — CHANGES_REQUESTED** [review]
Looks good! Thank you for your contribution.

Only a couple minor changes and this is good to go 🚀

**Comment by alexfertel** [comment]
Also, for this to take effect, you'd need to add this to every `Cargo.toml`

```toml
[lints]
workspace = true
```

see [the docs](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-lints-table) for more info.

**Comment by alexfertel** [comment]
Hey, could you please open the PR again from a different branch? PRs opened from the `main` branch of your repository cannot be edited by maintainers :)

I wanted to make the final adjustments before merging.

**Comment by developeruche** [comment]
> Hey, could you please open the PR again from a different branch? PRs opened from the `main` branch of your fork cannot be edited by maintainers :)
> 
> I wanted to make the final adjustments before merging.

Alright... on that now

**Comment by developeruche** [comment]
Would be closing this issue now; 
duplicate of #135

**Pull request closed without merging**
